adi/cn0554.py

```diff
@@ -0,0 +1,148 @@
+
+    _in_range_available = ["+/-13.75", "+27.5", "+2.5"]
+    _out_reference_available = [4.096, 2.5]
+    _out_channels = 16
+    _in_channels = 8
+
+    def __init__(self, uri="ip:analog.local"):
+        self.scale_res = 11  # Based on resistors used in input
+        self.adc = adi.ad7124(uri=uri)
+        self.adc._rx_unbuffered_data = False
+        self.adc._rx_output_type = "SI"
+        self.data_points = 1024
+        self.adc.sample_rate = 19200
+        self.adc_in_channels = []
+
+        for in_ch in self.adc.channel:
+            in_name = "voltage"
+            for char in in_ch.name:
+                if char.isnumeric():
+                    in_name += char
+
+            setattr(
+                self, in_name + "_in_range", self._in_range_available[0]
+            )  # default range of HW
+
+            self.adc_in_channels.append(in_name)
+
+        self.dac = adi.ltc2688(uri=uri)
+        self.dac_out_channels = self.dac.channel_names
+        self.out_reference = self._out_reference_available[0]
+
+    @property
+    def out_type(self):
+        """ Get value of CN0554's rx output type (raw or SI) """
+        return self.adc._rx_output_type
+
+    @out_type.setter
+    def out_type(self, val):
+        """ Set value of CN0554's rx output type (raw or SI) """
+        self.adc._rx_output_type = val
+
+    @property
+    def enabled_in_channels(self):
+        """ Get list of enabled input adc channels """
+        return self.adc.rx_enabled_channels
+
+    @enabled_in_channels.setter
+    def enabled_in_channels(self, val):
+        """ Set list of enabled input adc channels """
+        self.adc.rx_enabled_channels = val
+
+    @property
+    def out_channels(self):
+        """ Get number of DAC output channels """
+        return self._out_channels
+
+    @property
+    def in_channels(self):
+        """ Get number of ADC input channels """
+        return self._in_channels
+
+    @property
+    def out_reference(self):
+        """ Get voltage reference used for DAC channels """
+        return self.out_reference
+
+    @out_reference.setter
+    def out_reference(self, value):
+        """ Set voltage reference used for DAC channels """
+        if value in self._out_reference_available:
+            self.dac.vref = value
+
+    @property
+    def sample_rate(self):
+        """ Get CN0554's AD7124 sampling rate """
+        return self.adc.sample_rate
+
+    @sample_rate.setter
+    def sample_rate(self, value):
+        """ Set CN0554's AD7124 sampling rate """
+        self.adc.sample_rate = value
+
+    @property
+    def data_points(self):
+        """ Get number of datapoints used during capture """
+        return self.adc.rx_buffer_size
+
+    @data_points.setter
+    def data_points(self, value):
+        """ Set number of datapoints used during capture """
+        self.adc.rx_buffer_size = value
+
+    def get_data(self):
+        """ Get data from enabled ADC channels """
+        return self.adc.rx()
```

Comment:
This method should be called "rx"

Reply:
renamed function to rx in class and example code